### PR TITLE
Fix/reliability issues

### PR DESCRIPTION
**Pull request by lucidNTR**
@lucychambers @jo

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ downloadEmailAttachments({
   log: console.log,
   since: '2015-01-12',
   lastSyncIds: ['234', '234', '5345'] // ids already dowloaded and ignored, helpful because since is only supporting dates without time
-  attachmentHandler: function (attachmentData, callback) {
+  attachmentHandler: function (attachmentData, callback, errorCB) {
```
Comment (jo) [inline README.md:49]: Why don't you use the classic node style `callback(error, response)` here and instead add an extra `errorCB`?

Comment (lucidNTR) [inline README.md:49]: errrorCB can be called multiple times and aggregates all errors in one execution cycle. the callback may only be used once and marks the end of one function call. this design made debugging a lot easier. and is in this case an acceptable diversion from standard callbacks IMHOP

Comment (jo) [inline README.md:49]: I have concerns but lets talk about that after the demos ;)

Comment (lucidNTR) [inline README.md:49]: :-/

Comment (jo) [inline README.md:49]: Its hard to read for me :-/ I'm pretty sure you have good reasons but don't understand yet.

**Comment by jo** [comment]
I like it that the first commit updates the dependencies.

**Comment by lucidNTR** [comment]
nothing can get added to the queue unless at one point we will enable imap idle for push email. but that would be a completely new feature without the polling

**Comment by jo** [comment]
Given the queue only gets filled once - couldn't you use `async.parallel`? That way you would not have to deal with `drain` and `idle`

**Comment by lucidNTR** [comment]
parallel does not allow to specify a concurrency levenl, also queue works perfect now and will allow more flexible development of future features

**Comment by jo** [comment]
I meant `parallelLimit`

**Comment by jo** [comment]
:heavy_check_mark:  Tests pass

**Comment by jo** [comment]
:checkered_flag: works better!

I do _not see_ the error messages I repeately saw with the old version anymore:

```
timeout did not close imap connection in time
{
  "errors": [
    "connection closing in timeout error"
  ],
  "latestTime": "2015-12-05T08:32:00.000Z",
  "lastSyncId": 948,
  "uidvalidity": 1
}
```

**Comment by jo** [comment]
LGTM well done!

**Comment by lucidNTR** [comment]
puh

**Comment by lucidNTR** [comment]
parallelLimit is not documented, and i think less flexible...

**Comment by jo** [comment]
`parallelLimit(tasks, limit, [callback])`

**Pull request merged**